### DISCUSSION TITLE
PowerOnState - state my mqtt

### DISCUSSION THREAD

**Discussion by kleinmantara** [arendst/Tasmota#14935] [Q&A] [answered]
Hi,

it is posible to read the PowerOnState from mqtt? My shelly 1 with tasmota is not always on, when its start it shoud read the state from mqtt - maybe it is a new option for https://tasmota.github.io/docs/PowerOnState/ ?

**Comment by ascillato** [comment, marked as answer]
You can read all command by sending them without payload. So if you want to know what is the value for the setting called poweronstate, you just send:

Topic: cmnd/your_device_topic/poweronstate
Payload:

Payload must be empty. And the response will be the actual poweronstate value.

If what you want is to know the actual relay status (ON or OFF), you have to send the power command. Also you can check for that with the STATE command and with the teleperiod message.

**Reply by kleinmantara** [reply]
in other words:
- the shelly 1 is disconnect from the mains
- via mqtt i send the command cmnd/tasmota_switch/Power >ON
- the shelly 1 is connected to the mains
- reads the PowerOnState from mqtt cmnd/tasmota_switch/Power

**Reply by ascillato** [reply]
Mmmm, I think there is confusion here. Poweronstate and power are 2 different commands. Seems that what you want is just to use RETAINED MQTT messages. So as to do that, you don't need to configure anything in Tasmota. Just send by MQTT the `cmnd/tasmota_switch/Power ON` command RETAINED. Retained Mqtt messages are part of the MQTT protocol. When you send that, the Mqtt broker saves that in its own database and when the device is online, it will send that command.
Poweronstate is a configuration command of Tasmota, not related to MQTT. See the docs for more information on that.

**Comment by ascillato** [comment]
If what you want is that the control of your system to be centralized in your Mqtt broker, you have to use retained messages. This centralized control architecture is not recommended. What is recommended is the distributed control architecture as default. This has been discussed before several times, so for more details on that, please do a quick search in old issues. Thanks.

**Reply by kleinmantara** [reply]
Thanks the retain flag was the right hint.

> "This centralized control architecture is not recommended"

Of course, that depends very much on the use case, whether it makes sense or not.

**Reply by ascillato** [reply]
If you look in older issues you will see that it produces a lot of issues if not configured the whole system correctly and if your WiFi is not working properly.

For example for your setup: you send a retained message from your home automation software to turn on. That is saved in the Mqtt broker database and also sent to the device. The device has now the relay at ON state. Then you push the button of the device and now the relay is OFF. Then, your WiFi have a glitch and it disconnect and reconnects, so your broker sees an offline and online LWT so it will send again a power ON to your device and in that case is not what you want. And everytime your device reconnects from the wifi it will turn on the relay.
This happens when the architecture is mixed between distributed and centralized. Please, check old issues for further information on this matter.
